### PR TITLE
feat: icon in tab #47

### LINKED ISSUE #47: Add icon in tab

### PR DESCRIPTION
**Pull request by mizek1**
fix #47

### CODE REVIEW

File: projects/ion/src/lib/tab/tab.component.scss [hunk 1]
```diff
@@ -1,35 +1,64 @@
...
   cursor: pointer;
+  display: flex;
+  align-items: center;
+  gap: spacing(1);
+  @include icon-color($neutral-7);
 
   &:hover {
     color: $primary-5;
     background-color: $primary-1;
     border-color: $primary-3;
```
Comment (iurynogueira) [inline projects/ion/src/lib/tab/tab.component.scss:38]: @mixin make sense?

**Review by iurynogueira — CHANGES_REQUESTED** [review]

File: projects/ion/src/lib/tab/tab.component.scss [hunk 2]
```diff
@@ -1,35 +1,64 @@
 @import '../../styles/index.scss';
+
+@mixin icon-color($color) {
```
Comment (Deeved) [inline projects/ion/src/lib/tab/tab.component.scss:3]: I like it! :smile:

**Review by Deeved — APPROVED** [review]

File: projects/ion/src/lib/tab/tab.component.scss (hunk 1 above)
Comment (mizek1) [reply projects/ion/src/lib/tab/tab.component.scss:38]: Nice, I think mixin make sense here too!

**Comment by AlyssonMascarenhas** [comment]
Good job!

**Review by AlyssonMascarenhas — APPROVED** [review]

File: projects/ion/src/lib/tab/tab.component.spec.ts [hunk 3]
```diff
@@ -57,4 +56,10 @@ describe('TabComponent', () => {
     await sut({ label: defaultName, disabled: true });
     expect(screen.getByText(defaultName)).toBeDisabled();
   });
+
+  it('should render icon on tab', async () => {
+    await sut({ label: defaultName, iconType: 'trash' });
+    const elementRendered = document.getElementById('ion-icon-trash');
```
Comment (iurynogueira) [inline projects/ion/src/lib/tab/tab.component.spec.ts:62]: You can improve this returning `document.getElementById('ion-icon-trash');` in sut.

File: projects/ion/src/lib/tab/tab.component.spec.ts [hunk 4]
```diff
@@ -57,4 +56,10 @@ describe('TabComponent', () => {
     await sut({ label: defaultName, disabled: true });
     expect(screen.getByText(defaultName)).toBeDisabled();
   });
+
+  it('should render icon on tab', async () => {
+    await sut({ label: defaultName, iconType: 'trash' });
```
Comment (iurynogueira) [inline projects/ion/src/lib/tab/tab.component.spec.ts:61]:
```suggestion
    const elementRendered = await sut({ label: defaultName, iconType: 'trash' });
```

File: projects/ion/src/lib/tab/tab.component.spec.ts [hunk 5]
```diff
@@ -1,18 +1,17 @@
...
-}) => {
+const sut = async (customProps?: IonTabProps) => {
   await render(TabComponent, {
     componentProperties: customProps || {
       label: defaultName,
     },
+    imports: [CommonModule],
+    declarations: [IonIconComponent],
   });
 };
```
Comment (iurynogueira) [inline projects/ion/src/lib/tab/tab.component.spec.ts:16]:
```suggestion
const sut = async (customProps?: IonTabProps) => {
  await render(TabComponent, {
    componentProperties: customProps || {
      label: defaultName,
    },
    imports: [CommonModule],
    declarations: [IonIconComponent],
  });
  
  return document.getElementById('ion-icon-trash')
};
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]
🎉

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**